packages/api/src/controllers/animalBreedController.js

```diff
@@ -0,0 +1,43 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import AnimalBreedModel from '../models/animalBreedModel.js';
+
+const animalBreedController = {
+  getFarmAnimalBreeds() {
+    return async (req, res) => {
+      try {
+        const { farm_id } = req.headers;
+        const { type_id } = req.query;
```

Comment:
Curious about this decision to use query. We typically use param in url. 

Found this article that didn't find any best practices per se but had a point that query might be good for optional parameters and that path parameters were good for when they have subtrees.
https://stackoverflow.com/questions/4024271/rest-api-best-practices-where-to-put-parameters

Parent stack issue: https://stackoverflow.com/questions/25385559/rest-api-best-practices-args-in-query-string-vs-in-request-body

Reply:
I want to chat about this a little bit more tomorrow in tech daily, but yes, essentially both would be correct. The reason I ended up leaning towards a query param is I'm not sure if we'll ever need to fetch all breeds for all types for a farm at some point. If our URL reflects the nested parameters, such as `animal_type/1/animal_breed`, there wouldn't be a way to fetch without filtering by animal type. I don't know if we'll ever have a use case for that, but with that in mind I think the query params provide a little bit more flexibility, allowing us to filter by the parent entity or not depending on what we need.